fastapi/dependencies/utils.py

```diff
@@ -473,8 +473,9 @@ async def solve_dependencies(
     values: Dict[str, Any] = {}
     errors: List[ErrorWrapper] = []
     if response is None:
-        response = Response()
-        del response.headers["content-length"]
+        # HACK: avoid generating the content-length header by setting the status code
+        # to a number lower than 200.
+        response = Response(status_code=0)
```

Comment:
This is just changing one hack by another 🤷‍♂️

Reply:
But it's a more performant hack, because it does not require to create the mutable headers, just to remove the only entry.

I'm happy to remove it if that would mean the rest would be merged.

No idea though why it's creating an invalid response with a a `None` status code, which bypasses the type checks.

Reply:
@Kludex any thoughts? I'm happy to remove the hack if that would make the pr easier to review.

Reply:
Yes, I think that without this hack, the other part should be pretty straightforward once Sebastián gets around to review it. Great work on that part!

Reply:
Done, thanks!